site/docs/expressions/scalar_functions.md

```diff
@@ -1,52 +1,118 @@
 
+#### Value Argument Properties
 
+| Property | Description                                                  | Required                                                   |
+| -------- | ------------------------------------------------------------ | ---------------------------------------------------------- |
+| Name     | A human name for this argument to help clarify use.          | Optional, defaults to a name based on position (e.g. arg0) |
+| Type     | A fully concrete or parameterized (bound or not).            | Required                                                   |
+| Constant | Whether this argument is required to be a constant for invocation. For example, in some system a regular expression pattern would only be accepted as a literal and not a column value reference. | Optional, defaults to false                                |
 
-## Function Ids
+#### Type Argument Properties
 
-Each function signature is categorized based on a function signature id. The identifier includes two components:
+| Property | Description                                                  | Required                                                   |
+| -------- | ------------------------------------------------------------ | ---------------------------------------------------------- |
+| Type     | A partially or completely parameterized type. E.g. List&lt;K&gt; or K | Required                                                   |
+| Name     | A human name for this argument to help clarify use.          | Optional, defaults to a name based on position (e.g. arg0) |
 
-| Property        | Description                                                  |
-| --------------- | ------------------------------------------------------------ |
-| Organization Id | An unsigned 32 bit integer mapped to a list of known organizations listed in the [Substrait repository](https://github.com/substrait-io/substrait/blob/main/extensions/organizations.yaml). |
-| Function Id     | An unsigned 32 bit integer mapped to a list of known functions for the specific organization. For the Substrait organization, the function is listed in [here](https://github.com/substrait-io/substrait/blob/main/extensions/scalar_functions.yaml). |
+### Parameterized Types
 
-An organization is responsible for managing their own lists of functions. A function id should be used forever. If a function is deleted for any reason, the id should not be reused. Ideally each function signature will have one or more 
+Types are parametered by two types of values: by inner types (e.g. List&lt;K&gt;) and numeric values (e.g. VARCHAR(P,S)). Parameter names are simple strings (frequently a single character). There are two types of parameters: integer parameters and type parameters. 
```

Comment:
What about the names in a struct parameterization?  That doesn't seem to fit the dichotomy of types/integers.

Reply:
Yeah, there is no accomodation for that. For now, I would think that a plan builder would generate any required information using a type argument.